include/turtle/detail/type_name.hpp

```diff
@@ -33,7 +37,7 @@ namespace detail
     class type_name
     {
     public:
-        explicit type_name( const boost::detail::sp_typeinfo& info )
+        explicit type_name( const boost::core::typeinfo& info )
```

Comment:
Same here it probably depends on whether we included `boost/core` or `boost/detail`?

Reply:
Probably we could also just throw a `#if BOOST_VERSION` into the place here.. what do you think?